core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4206,6 +4206,55 @@ public static Map strToMap(String string, String stringDelimiter, String keyValu
     return map;
   }
 
+  /** Support the SUBSTRING_INDEX function. */
+  public static String substringIndex(String string, String delimiter, int count) {
+    if (string.isEmpty() || count == 0) {
```

Comment:
We should also add a test when the count argument is 0.

Reply:
done